src/qutip_cupy/dense_functions.py

```diff
@@ -29,3 +29,32 @@ def trace_cupydense(cp_arr):
     # @TODO: whnen qutip allows it we should remove this call to item()
     # as it takes a time penalty commmunicating data from GPU to CPU.
     return cp.trace(cp_arr._cp).item()
+
+
+hermdiff_kernel = cp.RawKernel(
+    r"""
+    #include <cupy/complex.cuh>
+    extern "C" __global__
+    void hermdiff(const complex<double>* x1,const int size,const double tol, bool* y) {
+        int tidx = blockDim.x * blockIdx.x + threadIdx.x;
+        int tidy = blockDim.y * blockIdx.y + threadIdx.y;
+        if((tidx < size) & (tidy < size)){
+            y[tidx+size*tidy] = norm(x1[tidx*size+tidy] - conj(x1[tidy*size+tidx])) < tol;
+        };
+    }""",
+    "hermdiff",
+)
+
+
+def isherm_cupydense(cp_arr, tol):
+    if cp_arr.shape[0] != cp_arr.shape[1]:
+        return False
+    size = cp_arr.shape[0]
+    diff = cp.empty((size, size), dtype=cp.bool_)
+    # TODO: check if there is a better way to set thread dim and block dim
+    block_size = 32
```

Comment:
Usually 128 or 256 is a better start point. 32 is only half of the warp size on AMD GPUs so we definitely want to try 64 and above for portability.